core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1817,6 +1842,9 @@ private void receive(Transaction tx, StoredBlock block, BlockChain.NewBlockType
             log.info("  <-pending");
 
         if (bestChain) {
+            boolean wasDead = dead.remove(txHash) != null;
```

Comment:
I don't fully understand this. DEAD means it's been double spent by something in the chain already. So processing a new block on the best chain should never resurrect a transaction. No?